src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -0,0 +1,205 @@
+    private _isDraggableOver: boolean = false;
+
+    set isDraggableOver(value: boolean) {
+        if (value) {
+            this.renderer.addClass(this.droppableEl, "draggable-over");
+        } else {
+            this.renderer.removeClass(this.droppableEl, "draggable-over");
+        }
+        this._isDraggableOver = value;
+    }
+
+    private _group: string|string[];
+    private _dropTolerance: ClrDropTolerance = {top: 0, right: 0, bottom: 0, left: 0};
+
+    @Input("clrGroup")
+    set group(value: string|string[]) {
+        this._group = value;
+    }
+
+    @Input("clrDropTolerance")
+    set dropTolerance(value: number|ClrDropTolerance) {
+        if (typeof value === "number") {
+            this._dropTolerance.top = value;
+            this._dropTolerance.right = value;
+            this._dropTolerance.bottom = value;
+            this._dropTolerance.left = value;
+        } else if (value) {
+            if (value.top) {
+                this._dropTolerance.top = value.top;
+            }
+            if (value.right) {
+                this._dropTolerance.right = value.right;
+            }
+            if (value.bottom) {
+                this._dropTolerance.bottom = value.bottom;
+            }
+            if (value.left) {
+                this._dropTolerance.left = value.left;
+            }
+        }
+    }
+
+    @Output("clrDragStart") dragStartEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragMove") dragMoveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnd") dragEndEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragLeave") dragLeaveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnter") dragEnterEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDrop") dropEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+
+    private unsubscribeFrom(subscription: Subscription): void {
+        if (subscription) {
+            subscription.unsubscribe();
+        }
+    }
+
+    private checkGroupMatch(draggableGroup: string|string[]): boolean {
+        if (!draggableGroup && this._group) {
+            return false;
+        }
+
+        if (!this._group && draggableGroup) {
+            return false;
+        }
+
+        if (!this._group && !draggableGroup) {
+            return true;
+        }
+
+        if (typeof draggableGroup === "string") {
+            if (typeof this._group === "string") {
+                return this._group === draggableGroup;
+            } else {
+                return this._group.indexOf(draggableGroup) > -1;
+            }
+        } else {
+            if (typeof this._group === "string") {
+                return draggableGroup.indexOf(this._group) > -1;
+            } else {
+                return (this._group as string[]).filter(groupKey => draggableGroup.indexOf(groupKey) > -1).length > 0;
```

Comment:
Nitpick: `array.filter(_ => condition).length > 0` is just `array.some(_ => condition)`

Reply:
CoOoOOol! Thanks :)